applications: nrf_desktop: Add buttons_sim documentation (#2025)
NordicPlayground/fw-nrfconnect-nrf: applications/nrf_desktop/doc/buttons_sim.rst

|                  |                  |                |                  | ``buttons_sim``                      |
|                  |                  |                |                  +--------------------------------------+
|                  |                  |                |                  | :ref:`hid_state`                     |
+------------------+------------------+----------------+------------------+--------------------------------------+

Configuration
*************

Simulated button sequence is triggered by ``button_event``.
You have to enable and configure the :ref:`buttons` module.

Set the ``CONFIG_DESKTOP_BUTTONS_SIM_ENABLE`` Kconfig option to enable the ``buttons_sim``  module.

Define the output key IDs sequence in the ``buttons_sim_def.h`` file located in board-specific directory in the ``configuration`` folder.
The mapping from the defined key ID to HID report ID and usage ID is defined in :file:`hid_keymap_def.h` (it may vary between boards).

Interval between subsequent simulated button presses is defined as ``CONFIG_DESKTOP_BUTTONS_SIM_INTERVAL``.
1 second is used by default.

> The simulated button sequence will be generated once by default.

Author: The sequence is not starting automatically when application is started. It has to be triggered with ``button_event``

> Set the ``CONFIG_DESKTOP_BUTTONS_SIM_LOOP_FOREVER`` to make sequence start from the beginning after it ends.

Author: i will change it to "make the sequence automatically resetart after it end"

Implementation details
**********************

The ``buttons_sim`` module generates button sequence using :c:type:`struct k_delayed_work`, that resubmits itself.

> Every call of work function submits press and release of single button from the sequence.

Reviewer: ```suggestion
Every work handler submits the press and the release of a single button from the sequence.
```

Author: There is one work handler that is called multiple times. I will change it here.

> Receiving ``button_event`` with key ID set to ``CONFIG_DESKTOP_BUTTONS_SIM_TRIGGER_KEY_ID`` stops generating the sequence (in case it is already being generated) or starts generating the sequence (otherwise).

Reviewer: ```suggestion
Receiving ``button_event`` with the key ID set to ``CONFIG_DESKTOP_BUTTONS_SIM_TRIGGER_KEY_ID`` either stops generating the sequence (in case it is already being generated) or starts generating it.
```

Configuration
*************

To configure the ``buttons_sim`` module:

1. Enable and configure the :ref:`buttons` module. ``button_event`` is used to trigger the simulated button sequence.
#. Enable the ``buttons_sim``  module by setting the ``CONFIG_DESKTOP_BUTTONS_SIM_ENABLE`` Kconfig option.
#. Define the output key ID sequence in the :file:`buttons_sim_def.h` file located in the board-specific directory in the :file:`configuration` folder. The mapping from the defined key ID to the HID report ID and usage ID is defined in :file:`hid_keymap_def.h` (this might be different for different boards).

> #. Define the interval between subsequent simulated button presses - ``CONFIG_DESKTOP_BUTTONS_SIM_INTERVAL``. 1 second is used by default.

Reviewer: ```suggestion
#. Define the interval between subsequent simulated button presses - ``CONFIG_DESKTOP_BUTTONS_SIM_INTERVAL``. One second is used by default.
```

If you want the sequence to automatically restart after it ends, set ``CONFIG_DESKTOP_BUTTONS_SIM_LOOP_FOREVER``.

> By default the sequence is generated only once.

Reviewer: ```suggestion
By default, the sequence is generated only once.
```
